### ISSUE TITLE
API Overhaul: Rename 'widget' to 'field type'

### ISSUE THREAD

**Issue by austincondiff** [netlify/netlify-cms#3719]
As part of the API overhaul effort per #1041  and #1409, I propose we move away from the somewhat ambiguous term “widget” in reference to field types within the editor. The term widget can be synonymous with sections within a dashboard and as we plan on introducing a dashboard view into our UI, it may be a bit confusing, especially for first-time users. 

We currently call them fields within the config file. So I'd imagine it may be more clear if we changed widget to type in each field in the fields array to avoid confusion in the future.

Likewise instead of `registerWidget` we might rename this `registerFieldType` making way for `registerDashboardWidget`.

**Comment by jeremyzilar** [comment]
Yes 💯 it took a bit to let our editors know that the widgets in the docs were equal to the fields settings in the `config.yml`. Making the naming more consistent between the config settings and the docs and the APIs would be super great.

Related: https://github.com/netlify/netlify-cms/issues/3624
